src/components/userFeedback/UserFeedback.spec.tsx

```diff
@@ -0,0 +1,143 @@
+  const defaultPropsWithDisplay: IUserFeedbackProps = generateProps('', '', '', 'block');
+
+  const getShallowOutput = (props: IUserFeedbackProps): ShallowWrapper<IUserFeedbackProps, any> => {
+    return shallow(<UserFeedback {...props} />);
+  };
+
+  describe('"Shallow rendered" component', () => {
+    describe('rendering', () => {
+      it('should render without errors', () => {
+        expect(() => getShallowOutput(defaultProps)).not.toThrow();
+        expect(() => getShallowOutput(defaultPropsWithDisplay)).not.toThrow();
+        expect(() => getShallowOutput(generateProps('', '', ''))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.VALID, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.WARNING, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.ERROR, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.VALID, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.WARNING, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.ERROR, 'm1', 'block'))).not.toThrow();
+      });
+    });
+
+    describe('text content', () => {
+      it('should contain the exact text passed to prop feedbackText', () => {
+        let testText = 'testing';
+        let wrongText: string;
+
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).toBe(testText);
+
+        testText = 'hello this is a longer test string over here';
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).toBe(testText);
+
+
+        testText = 'hello this is a test over here';
+        wrongText = 'hello this is the wrong text over here';
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).not.toBe(wrongText);
+      });
+    });
+
+    describe('style for each state', () => {
+      describe('non existent state', () => {
+        it('should be invisible if state provided does not exist', () => {
+          let nonExistentState = 'NON_EXISTENT_STATE';
+          let emptyState = '';
+
+          expect(getShallowOutput(generateProps('', nonExistentState, '')).hasClass('hidden')).toBe(true);
+          expect(getShallowOutput(generateProps('', emptyState, '')).hasClass('hidden')).toBe(true);
+        });
+      });
+
+      describe('VALID state', () => {
+        it('should be invisible on state VALID without displayOnShow prop', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.VALID, '')).hasClass('hidden')).toBe(true);
+        });
+
+        it('should be invisible on state VALID, even with prop displayOnShow provided', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.VALID, '')).hasClass('hidden')).toBe(true);
```

Comment:
_should be invisible on state VALID without displayOnShow prop_ and _should be invisible on state VALID, even with prop displayOnShow provided_ are doing the exact same thing. Are you sure you second test is valid?